tools/istio-nftables/pkg/README.md

```diff
@@ -0,0 +1,132 @@
+for working with nftables.
+
+```sh
+table inet istio-proxy-nat {
+    chain prerouting {
+        type nat hook prerouting priority dstnat; policy accept;
+        # Istio redirection rules...
+    }
+
+    chain output {
+        type nat hook output priority dstnat; policy accept;
+        # Istio output rules...
+    }
+
+    # Custom Istio chains for traffic processing
+    chain istio-inbound { }
+    chain istio-output { }
+    chain istio-redirect { }
+    # ...additional chains
+}
+```
+
+## Configuration
+
+The `nftables` backend is **disabled by default**. To enable it, you must set the following value during installation process
+
+```sh
+--set values.global.nativeNftables=true
+```
+
+This flag configures Istio to use the `nftables` backend instead of `iptables` for traffic redirection.
+
+## Implementation Details
+
+### Table Structure
+
+The `nftables` backend creates three main tables in the `inet` family:
+
+- **`istio-proxy-nat`**: Handles traffic redirection using DNAT/REDIRECT targets
+- **`istio-proxy-mangle`**: Manages packet marking and TPROXY operations
+- **`istio-proxy-raw`**: Handles connection tracking zones for DNS traffic
+
+### Chain Organization
+
+Each table contains both netfilter hook chains and custom chains:
+
+**Hook/Base Chains** (attach to kernel netfilter hooks):
+- `prerouting`: Processes incoming packets before routing decisions
+- `output`: Processes locally-generated packets
+
+**Custom Chains** (called via jump rules):
+- `istio-inbound`: Inbound traffic processing logic
+- `istio-output`: Outbound traffic processing logic
+- `istio-redirect`: Common redirection target for Envoy
+- `istio-tproxy`: TPROXY-specific handling
+- `istio-divert`: Traffic diversion based on packet marking
+
+### Rule Application Process
+
+1. **Rule Generation**: The `NftablesRuleBuilder` constructs rules based on Istio configuration
+1. **Transaction Creation**: All rules are assembled into a single `knftables.Transaction`
+1. **Atomic Application**: The entire transaction is applied atomically via the `knftables` library, which internally validates rule structure and uses `nft -f -` for execution
+1. **Error Handling**: On failure, no partial state is left behind
+
+This ensures that the traffic redirection is never in an inconsistent state during updates.
+
+## Reconciliation Logic
+
+The implementation generates a **complete `nftables` ruleset** in memory and then applies it as a single, atomic transaction using knftables APIs.
+It **does not** patch or modify individual rules on the fly. This ensures the ruleset is always consistent and valid.
+The implementation also ensures that old rules are cleaned up and new rules are created at the same time, avoiding any partial updates.
+
+## Debugging Guidelines
+
+The following `nft` commands are useful for troubleshooting. The implementation includes `counters` on all rules to provide useful debugging information.
```

Comment:
Is there a plan to also document this on istio.io? Might be useful to have there as well as most users won't look in Github for these steps

Reply:
Yes @dhawton. Please see my comment above - https://github.com/istio/istio/pull/57275/files#r2257765755